flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -232,17 +177,10 @@ private TestingDispatcher createAndStartDispatcher(
 
     @After
     public void tearDown() throws Exception {
+        super.tearDown();
```

Comment:
I think this should go after the if block because the dispatcher might still use some of the services terminated in `super.tearDown()`.